src/json.ts

```diff
@@ -0,0 +1,217 @@
+                break;
+            }
+
+            let code = next();
+
+            if (code === CharCode.doubleQuote) {
+                // End of string. We're done
+                break;
+            }
+
+            if (code === CharCode.backSlash) {
+                // Skip escaped character. We don't care about verifying the escape sequence.
+                // We just don't want to accidentally scan an escaped double-quote as the end of the string.
+                index++;
+            }
+
+            if (isLineBreak(code)) {
+                // string ended unexpectedly
+                break;
+            }
+        }
+    }
+
+    while (true) {
+        let code = next();
+
+        switch (code) {
+            // byte-order mark
+            case CharCode.byteOrderMark:
+                // We just skip the byte-order mark
+                parts.push(text.substring(partStart, index - 1));
+                partStart = index;
+
+            // strings
+            case CharCode.doubleQuote:
+                scanString();
+                break;
+
+            // comments
+            case CharCode.slash:
+                // Single-line comment
+                if (peek() === CharCode.slash) {
+                    // Be careful not to include the first slash in the text part.
+                    parts.push(text.substring(partStart, index - 1));
+
+                    // Start after the second slash and scan until a line-break character is encountered.
+                    index++;
+                    while (index < length) {
+                        if (isLineBreak(peek())) {
+                            break;
+                        }
+
+                        index++;
+                    }
+
+                    partStart = index;
+                }
+
+                // Multi-line comment
+                if (peek() === CharCode.asterisk) {
+                    // Be careful not to include the first slash in the text part.
+                    parts.push(text.substring(partStart, index - 1));
+
+                    // Start after the asterisk and scan until a */ is encountered.
+                    index++;
+                    while (index < length) {
+                        if (peek() === CharCode.asterisk && peek(1) === CharCode.slash) {
+                            index += 2;
+                            break;
+                        }
+
+                        index++;
+                    }
+
+                    partStart = index;
+                }
+
+                break;
+
+            case CharCode.comma:
+                // Ignore trailing commas in object member lists and array element lists
+                let nextCode = peekPastWhitespace();
+                if (nextCode === CharCode.closeBrace || nextCode === CharCode.closeBracket) {
+                    parts.push(text.substring(partStart, index - 1));
+                    partStart = index;
```

Comment:
`peekPastWhitespace` doesn't do anything but look ahead. It doesn't modify `index`